pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -468,11 +468,33 @@ func (cm *certificateManager) getKeyPair(cli client.Client, secretName, secretNa
 		}
 
 		if timeInvalid {
-			return nil, nil, fmt.Errorf("secret %s/%s is not valid at this date", secretNamespace, secretName)
+			if readCertOnly {
+				cm.log.Info("secret %s/%s is not valid at this date, so we return nil and continue as though it does not exist", secretNamespace, secretName)
+				return nil, nil, nil
+			}
+			return nil, nil, fmt.Errorf("secret %s/%s is not valid at this date for a certificate not created by the operator, user action required", secretNamespace, secretName)
 		}
 		return nil, nil, newCertExtKeyUsageError(secretName, secretNamespace, requiredKeyUsages)
 	}
 
+	if !readCertOnly && x509Cert.NotAfter.Before(time.Now().Add(30*24*time.Hour)) {
+		// The certificate is not valid one month from now. Let's start the rotation process, so there will be plenty of time
+		// to roll out the changes without disruption. All components that need to trust this certificate are already
+		// trusting the issuer, so there will be no disruption.
+		if !strings.HasPrefix(x509Cert.Issuer.CommonName, rmeta.TigeraOperatorCAIssuerPrefix) {
+			cm.log.Info("this certificate will soon expire and is not managed by the operator, user action required", "name", secretName)
+		} else {
+			if cm.keyPair.CertificateManagement != nil {
+				// When certificate management is enabled, we can simply return a certificate management key pair;
+				// the old secret will be deleted automatically.
+				return certificateManagementKeyPair(cm, secretName, secretNamespace, dnsNames), nil, nil
+			}
+			cm.log.Info("this certificate will soon expire, so we rotate it now", "name", secretName)
```

Comment:
Nit, this seems a little word, could just say "Rotating soon to expire certificate <name>."